lib/interceptor.js

```diff
@@ -11,12 +11,20 @@ var interceptor = {
       polyfillFormDataEntries();
     }
 
-    if (window.sessionStorage && window.sessionStorage.removeItem) {
+    if (supportsSessionStorage()) {
       window.sessionStorage.removeItem(NAMESPACE);
     }
 
     if (typeof window.fetch == 'function') {
       replaceFetch();
+      if (
+        typeof window.Promise === 'undefined' ||
```

Comment:
Is there ever a case where `window.fetch` exists, but `window.Promise` does not? I can only imagine very weird (and broken!) cases of polyfilling.

Reply:
There _shouldn't_, but I could imagine that someone might polyfill `fetch`, `Promise.then`, and `Promise.catch`, but not polyfill `Promise.all`. I figured it was better that we detect this scenario and throw promptly, rather than fail when the fetch wrapper is invoked because of this library's use of `Promise.all`